lib/src/sdk_api/view/video_view.dart

```diff
@@ -275,20 +297,28 @@ class _VideoViewState extends State<VideoView> {
   }
 
   void _updateParticipantAndStream() {
-    final participant = _participant;
-    final mediaStream = widget.mediaStream;
-    final mediaStreamSelector = widget.mediaStreamSelector;
-    if (participant != null) {
-      if (mediaStreamSelector != null) {
-        setState(() {
-          _participant = participant;
-          _mediaStream = mediaStreamSelector(participant.streams);
-        });
-      } else if (mediaStream != null) {
-        setState(() {
-          _participant = participant;
-          _mediaStream = mediaStream;
-        });
+    final controller = _controller;
+    if (controller != null) {
+      setState(() {
+        _participant = controller._participant;
+        _mediaStream = controller._mediaStream;
+      });
+    } else {
+      final participant = _participant;
+      final mediaStream = widget.mediaStream;
+      final mediaStreamSelector = widget.mediaStreamSelector;
+      if (participant != null) {
+        if (mediaStreamSelector != null) {
+          setState(() {
+            _participant = participant;
+            _mediaStream = mediaStreamSelector(participant.streams);
+          });
+        } else if (mediaStream != null) {
+          setState(() {
+            _participant = participant;
+            _mediaStream = mediaStream;
+          });
+        }
```

Comment:
VideoView class contains too many contructor. This cause that anybody that try to use it it will be confused. I think this class must be redesing because right now there is a big mess